pkg/service/service.go

```diff
@@ -0,0 +1,105 @@
+package service
+
+import (
+	"github.com/pkg/errors"
+	applabels "github.com/redhat-developer/odo/pkg/application/labels"
+	componentlabels "github.com/redhat-developer/odo/pkg/component/labels"
+	"github.com/redhat-developer/odo/pkg/occlient"
+	"github.com/redhat-developer/odo/pkg/util"
+)
+
+// ServiceInfo holds all important information about one service
+type ServiceInfo struct {
+	Name   string
+	Type   string
+	Status string
+}
+
+// ListCatalog lists all the available service types
+func ListCatalog(client *occlient.Client) ([]string, error) {
+
+	clusterServiceClasses, err := client.GetClusterServiceClassExternalNames()
+	if err != nil {
+		return []string{}, errors.Wrapf(err, "unable to get cluster serviceClassExternalName")
+	}
+
+	return clusterServiceClasses, nil
+}
+
+// CreateService create new service from serviceCatalog
+func CreateService(client *occlient.Client, serviceName string, serviceType string, applicationName string) error {
+	labels := componentlabels.GetLabels(serviceName, applicationName, true)
+	// save service type as label
+	labels[componentlabels.ComponentTypeLabel] = serviceType
+	err := client.CreateServiceInstance(serviceName, serviceType, labels)
+	if err != nil {
+		return errors.Wrap(err, "unable to create service instance")
+
+	}
+	return nil
+}
+
+// DeleteService will delete the service with the provided `name`
+func DeleteService(client *occlient.Client, name string, applicationName string, projectName string) error {
+
+	labels := componentlabels.GetLabels(name, applicationName, false)
+	err := client.DeleteServiceInstance(labels)
+	if err != nil {
+		return errors.Wrapf(err, "unable to retrieve list of services")
+	}
+	return nil
+
+}
+
+// List lists all the deployed services
+func List(client *occlient.Client, applicationName string, projectName string) ([]ServiceInfo, error) {
+	labels := map[string]string{
+		applabels.ApplicationLabel: applicationName,
+	}
+
+	applicationSelector := util.ConvertLabelsToSelector(labels)
+	serviceInstanceList, err := client.GetServiceInstanceList(projectName, applicationSelector)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to list services")
+	}
+
+	var services []ServiceInfo
+	for _, elem := range serviceInstanceList {
+		services = append(services, ServiceInfo{Name: elem.Labels[componentlabels.ComponentLabel], Type: elem.Labels[componentlabels.ComponentTypeLabel], Status: elem.Status.Conditions[0].Reason})
+	}
+
+	return services, nil
```

Comment:
above code needs comments explain what they do. 

For the first code block, explain why you're grabbing labels.

For second code block, something like this would suffice:
`// Iterate through serviceInstanceList and add to service`.

Reply:
added